File: jpa-test/src/main/java/zh/tools/test/jpa/UserService.java
package zh.tools.test.jpa;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import zh.tools.jpa.BaseFilterService;
import zh.tools.jpa.BaseRepositorySupport;

@Service
public class UserService extends BaseFilterService<User, Long> {

    @Autowired
    private UserRepository userRepository;

    @Override
    public BaseRepositorySupport<User, Long> repository() {
        return userRepository;
    }
}

File: mybatis-plus-test/src/main/java/zh/tools/test/mybatisplus/UserService.kt
package zh.tools.test.mybatisplus

import org.springframework.stereotype.Service
import zh.tools.maptosql.JsonToSqlConverter
import zh.tools.maptosql.MySqlDialect
import zh.tools.mybatisplus.CommonService

val jsonToSqlConverter = JsonToSqlConverter(MySqlDialect())

@Service
class UserService : CommonService<UserMapper, User, Long>(
)
